### DISCUSSION TITLE
Overview Hackathon Projects

### DISCUSSION THREAD

**Discussion by daurer** [ptycho/ptypy#573] [Projects]
Here is of projects that are being explored as part of the hackathon during the PtyPy workshop 2024!

**If you are a participant of the hackathon, please comment below and let us know what you would like to work on!**
You could also create an [issue](https://github.com/ptycho/ptypy/issues/new/choose) and provide the reference so that we can add your idea/project/problem to the tables below.

## Coding Projects

| Name                                              | What's the project?            | Supervisor                 | References (Issues, PRs) |
| :--------------------------------------------- | :--------------------------------- |  :-------------------------- | :----------------------------------|
| Letizia Protopapa (@Letizia97 )      | 3D ML solver                     | Pierre Thibault           |                                          |  
| Jaroslav Fowkes (@jfowkes )          |                                           |                                   |                                          |
| Alan Muriithi                                    |                                           |                                   |                                          |
| Francesco Guzzi (@frgfrg )             |                                           |                                   |                                          |
| Jaroslav Fowkes (@jfowkes )          | Merge LBFGS                   |  Timothy Poon            |   #485                               |
| Jaroslav Fowkes (@jfowkes )          | Get LSQ-ML working        | Pierre Thibault            |  #574                               |
| Jaroslav Fowkes (@jfowkes )          | Gradient-based pos ref     | Pierre Thibault            |                                          |
| Tang Li (@ltang320 )                       | Accelerate OPR                | Timothy Poon             |    #534                              |
| Maik Kahnt (@kahntm )                  | Refractive index ptycho     | Pierre Thibault            |                                          | 
| Julio da Silva  (@jcesardasilva )     | Super-resolution feature    | Bjoern Enders            |   #577                                 |
| Rebecka Lexelius  (@Lexelius )      | Streaming?                        | Bjoern Enders            |                                          |
| Damian Guenzing                            | Improve documentation    |  Benedikt Daurer         |                                          |
|                                                         | Incoherent Background     |  Benedikt Daurer        | #341                                 |
|                                                         | Accelerate Multi-slice         |                                    |                                          |
|  Amirafshar  Moshtaghpour  (@Amirafsharm ) | Better UI/UX for electron ptycho | Benedikt Daurer  |                                          |

## Working on Data

| Name                                |   Challenge/Issue                                           | Supervisor                 | References (Issues, PRs) |
| :--------------------------------- | :-----------------------------------------------------------|  :-------------------------- | :----------------------------------|
| Mikhail Lyubomirskiy         | Strength of separate probe update                 | Pierre Thibault           |                                          |
| Tang Li (@ltang320 )        | Dataset with slowly varying probe                   | Pierre Thibault           |                                          |
| Damian Guenzing             | Reflection ptychography data                          | Bjoern Enders           |                                          |
| Yuri Rossi Tonin  (@yurirtonin ) |                                                                         | Bjoern Enders           |                                          |
| Mauro Luiz Brandão Junior |                                                                       | Bjoern Enders           |                                          |
| Marco Margini                    | Nearfield ptychography from ESRF               | Benedikt Daurer       |                                          |
| Julio da Silva (@jcesardasilva ) | Testing multislice on Nearfield ptycho data |                          |                                          |

## Installing/Using PtyPy

| Name                            |  Learning target              | Supervisor              |
| :-----------------------------  | :------------------------------- | :------------------------- |
| Jakata Kudakwashe      | Basics of ptypy              | Benedikt Daurer       |
| Alan Muriithi                  | Basics of ptypy               | Benedikt Daurer      |
| Yuri Rossi Tonin            | Basics of ptypy               | Benedikt Daurer      |
| Mauro Luiz Brandão Junior | Basics of ptypy         | Benedikt Daurer      |
| Matteo Bille                   | Basics of ptypy               | Benedikt Daurer      |
| Jean-Luc Rouviere        | Using ptypy for electron ptychography | Benedikt Daurer |
| Alexis Wartelle               | Using ptypy for electron ptychography | Benedikt Daurer |
| Damian Guenzing         | Deeper functionalities     | Bjoern Enders         |

**Comment by yurirtonin** [comment]
Hi everyone! Mauro and I talked, we would like to switch our projects to "Analyst data" and "Install/using PtyPy". Working on code just in case there's some time left. Cheers!
